modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/SiteResourceImpl.java

```diff
@@ -49,12 +51,16 @@
 	public Page<Site> getMyUserAccountSitesPage(Pagination pagination)
 		throws Exception {
 
+		List<Group> userSitesGroups = _groupLocalService.getUserSitesGroups(
+			contextUser.getUserId());
+
 		return Page.of(
 			transform(
 				_groupService.getUserSitesGroups(
 					contextUser.getUserId(), pagination.getStartPosition(),
 					pagination.getEndPosition()),
-				this::_toSite));
+				this::_toSite),
+			pagination, userSitesGroups.size());
```

Comment:
Maybe it is better to use _groupLocalService.getUserSitesGroupsCount to get the total count, could be?

Calling two times to getUserSitesGroups seems to me that we are hitting the database two times to get almost the same info and could cause impact performance.

Other option is trying to find a mechanism that allow us to get the all the items to get the total count and then select the ones desired by the pagination info